app/models/concerns/with_organization_status.rb

```diff
@@ -0,0 +1,26 @@
+module WithOrganizationStatus
+
+  def status
+    if disabled?
+      @status ||= WithOrganizationStatus::Disabled.new self
```

Comment:
The repeated assignment can be moved outside the if, or - even better - the if could me moved to a separate method